### PR TITLE
maintenance to use discovery update

### PR DESCRIPTION
**Pull request by nao-pon**
This fix has not been verified yet and requires user verification.

### CODE REVIEW

File: custom_components/tuya_custom/tuyaha/devices/climate.py [hunk 1]
```diff
@@ -65,21 +65,22 @@ def max_humidity(self):
...
 
     def set_humidity(self, humidity):
         """Set new target humidity."""
         raise NotImplementedError()
 
     def set_fan_mode(self, fan_mode):
         """Set new target fan mode."""
-        self.api.device_control(self.obj_id, "windSpeedSet", {"value": fan_mode})
+        if self._control_device("windSpeedSet", {"value": fan_mode}):
+            self._update_data("windspeed", fan_mode)
```
Comment (ollo69) [inline custom_components/tuya_custom/tuyaha/devices/climate.py:84]: I think that here you will receeive "low, medium, high" and should be converted to "1,2,3" (inverted of convertion done in current_fan_mode method)

File: custom_components/tuya_custom/tuyaha/devices/climate.py [hunk 2]
```diff
@@ -110,7 +111,12 @@ def support_humidity(self):
             return False
 
     def turn_on(self):
-        self.api.device_control(self.obj_id, "turnOnOff", {"value": "1"})
+        if self._control_device("turnOnOff", {"value": "1"}):
+            self._update_data("state", True)
```
Comment (ollo69) [inline custom_components/tuya_custom/tuyaha/devices/climate.py:121]: I think that here is used "true" based on original library

File: custom_components/tuya_custom/tuyaha/devices/climate.py [hunk 3]
```diff
@@ -110,7 +111,12 @@ def support_humidity(self):
...
 
     def turn_off(self):
-        self.api.device_control(self.obj_id, "turnOnOff", {"value": "0"})
+        if self._control_device("turnOnOff", {"value": "0"}):
+            self._update_data("state", False)
```
Comment (ollo69) [inline custom_components/tuya_custom/tuyaha/devices/climate.py]: I think that here is used "false" based on original library

File: custom_components/tuya_custom/tuyaha/devices/fan.py [hunk 4]
```diff
@@ -17,20 +17,27 @@ def oscillating(self):
...
         else:
             command = "swingClose"
-        self.api.device_control(self.obj_id, command)
+        if self._control_device(command):
+            self._update_data("direction", oscillating)
 
     def turn_on(self):
-        self.api.device_control(self.obj_id, "turnOnOff", {"value": "1"})
+        if self._control_device("turnOnOff", {"value": "1"}):
+            self._update_data("state", True)
```
Comment (ollo69) [inline custom_components/tuya_custom/tuyaha/devices/fan.py]: same as climate

File: custom_components/tuya_custom/tuyaha/devices/fan.py [hunk 5]
```diff
@@ -17,20 +17,27 @@ def oscillating(self):
...
 
     def turn_off(self):
-        self.api.device_control(self.obj_id, "turnOnOff", {"value": "0"})
+        if self._control_device("turnOnOff", {"value": "0"}):
+            self._update_data("state", False)
```
Comment (ollo69) [inline custom_components/tuya_custom/tuyaha/devices/fan.py]: same as climate

**Review by ollo69 — CHANGES_REQUESTED** [review]

File: custom_components/tuya_custom/tuyaha/devices/climate.py (hunk 1 above)
Comment (nao-pon) [reply custom_components/tuya_custom/tuyaha/devices/climate.py:84]: OK, I'm not sure if the parameters given here are "1", "2", "3" or "low", "medium", "high", so I'll change it as follows.

```python
        if self._control_device("windSpeedSet", {"value": fan_mode}):
            fanList = self.fan_list()
            if fan_mode in fanList:
                val = str(fanList.index(fan_mode) + 1)
            else:
                val = fan_mode
            self._update_data("windspeed", val)
```

File: custom_components/tuya_custom/tuyaha/devices/climate.py (hunk 2 above)
Comment (nao-pon) [reply custom_components/tuya_custom/tuyaha/devices/climate.py:121]: OK, I'll fix it.

File: custom_components/tuya_custom/tuyaha/devices/climate.py (hunk 3 above)
Comment (nao-pon) [reply custom_components/tuya_custom/tuyaha/devices/climate.py]: 👍

File: custom_components/tuya_custom/tuyaha/devices/fan.py (hunk 4 above)
Comment (nao-pon) [reply custom_components/tuya_custom/tuyaha/devices/fan.py]: 👍

File: custom_components/tuya_custom/tuyaha/devices/fan.py (hunk 5 above)
Comment (nao-pon) [reply custom_components/tuya_custom/tuyaha/devices/fan.py]: 👍

**Comment by nao-pon** [comment]
@ollo69 Thanks! 👍

**Pull request merged**